docs/ref/models/database-functions.txt

```diff
@@ -293,3 +293,518 @@ Usage example::
+    ...    start_time__second=ExtractSecond('start_time')).count()
+    1
+
+``Trunc``
+---------
+
+.. class:: Trunc(date_expression, kind, output_field=None, **extra)
+
+Truncates a date up to a significant component.
+
+When you only care if something happened in a particular year, hour, or day,
+but not the exact moment in time, then ``Trunc`` (and its subclasses) can be
+useful to filter or group your data. Calculating the number of sales per day
+is a common use case for ``Trunc``.
+
+``Trunc`` takes a single ``date_expression`` representing a ``DateField`` or
+``DateTimeField``, a ``kind`` representing a date part, and an ``output_field``
+that's either ``DateTimeField()`` or ``DateField()``. It returns a datetime or
+date, depending on ``output_field``, with fields up to ``kind`` set to their
+minimum value.
+
+Given the datetime `2015-06-15 14:30:50.321` the ``kind``\s supported by
+Django will return:
+
+* "year": 2015-01-01 00:00:00.000
+* "month": 2015-06-01 00:00:00.000
+* "day": 2015-06-15 00:00:00.000
+* "hour": 2015-06-15 14:00:00.000
+* "minute": 2015-06-15 14:30:00.000
+* "second": 2015-06-15 14:30:50.000
+
+The ``kind``\s above all have corresponding ``Trunc`` subclasses that should be
+preferred in the majority of cases. The subclasses below are all defined as
+transforms, but they are not registered with any fields by default.
+
+Usage example::
+
+    >>> from datetime import datetime
+    >>> from django.db.models import Count, DateTimeField
+    >>> from django.db.models.functions import Trunc
+    >>> start1 = datetime(2015, 6, 15, 14, 30, 50, 321)
+    >>> start2 = datetime(2015, 6, 15, 14, 40, 2, 123)
+    >>> start3 = datetime(2015, 12, 25, 10, 5, 27, 999)
+    >>> Experiment.objects.create(start_time=start1)
+    >>> Experiment.objects.create(start_time=start2)
+    >>> Experiment.objects.create(start_time=start3)
+    >>> experiments_per_day = Experiment.objects.annotate(
+    ...    start_day=Trunc('start_time', 'day', output_field=DateTimeField())
+    ... ).values('start_day').annotate(experiments=Count('id'))
+    >>> for exp in experiments_per_day:
+    ...     print(exp['start_day'], exp['experiments'])
+    ...
+    2015-06-15 00:00:00 2
+    2015-12-25 00:00:00 1
+    >>> experiments = Experiment.objects.annotate(
+    ...    start_day=Trunc('start_time', 'day', output_field=DateTimeField())
+    ... ).filter(start_day=datetime(2015, 6, 15))
+    >>> for exp in experiments:
+    ...     print(exp.start_time)
+    ...
+    2015-06-15 14:30:50.000321
+    2015-06-15 14:40:02.000123
+
+``TruncYear``
+-------------
+
+.. class:: TruncYear(date_expression, **extra)
+
+    .. attribute:: kind = 'year'
+    .. attribute:: output_field = DateField()
+
+Logically equivalent to
+``Trunc('date_field', 'year', output_field=DateField())``. It truncates all
+parts of the date up to the year and allows grouping or filtering data within
+the same year. The return type is always ``DateField``.
```

Comment:
That's bad when `USE_TZ = True` and one is truncating datetimes. In that case the result depends on the current time zone in effect at the time the query ran. Returning dates (instead of datetimes) looses that information.